### PR TITLE
Increased test coverage for use-resize-observer

### LINKED ISSUE #2079: Enhance Test Coverage for `@yamada-ui/use-resize-observer`
package: @yamada-ui/use-resize-observer

### Description
`@yamada-ui/use-resize-observer` has a test coverage of less than 95%. We will enhance it to above 95%.

### Target files
- [packages/hooks/use-resize-observer/src/index.ts](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/hooks/use-resize-observer/src/index.ts)
  - [L57](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/hooks/use-resize-observer/src/index.ts#L57)

### Problem Statement/Justification
Testing is essential to ensure quality when operating a package.

### Additional Information
While it's ideal to cover all tests, if there are too many lines to work on, please create a PR within your capacity.
Any deficiencies not covered by the PR will be automatically created as issues.

### PR DESCRIPTION
**Pull request by kurakke**
Closes #2079

## Description
Increased test coverage.
https://app.codecov.io/gh/yamada-ui/yamada-ui/pull/2139/blob/packages/hooks/use-resize-observer/src/index.ts

## Is this a breaking change (Yes/No):
No

### CODE REVIEW

File: packages/hooks/use-resize-observer/tests/index.test.tsx [hunk 1]
```diff
@@ -45,6 +39,47 @@ describe("useResizeObserver", () => {
     render(<ButtonWithSize />)
     const button = await screen.findByRole("button")
 
-    await waitFor(() => expect(button.textContent).toBe("400 x 320"))
```
Comment (108yen) [inline packages/hooks/use-resize-observer/tests/index.test.tsx:48]: Dont remove `waitFor`.

File: .changeset/empty-rules-laugh.md
Comment (108yen) [inline .changeset/empty-rules-laugh.md:1]: Please remove this file.

File: packages/hooks/use-resize-observer/tests/index.test.tsx [hunk 2]
```diff
@@ -45,6 +39,47 @@ describe("useResizeObserver", () => {
...
+    return (
+      <Button ref={ref}>
+        {width} x {height}
+      </Button>
+    )
+  }
+
+  test("returns width and height correctly", async () => {
+    render(<ButtonWithSize />)
+    const button = await screen.findByText("150 x 200")
```
Comment (108yen) [inline packages/hooks/use-resize-observer/tests/index.test.tsx:81]:
```tsx
const button = await screen.findByRole("button")
```

File: packages/hooks/use-resize-observer/tests/index.test.tsx [hunk 3]
```diff
@@ -1,10 +1,8 @@
 import { Button } from "@yamada-ui/react"
-import { render, screen, waitFor } from "@yamada-ui/test"
-import { useResizeObserver } from "../src"
+import { render, screen } from "@yamada-ui/test"
+import { useElementSize, useResizeObserver } from "../src"
 
 describe("useResizeObserver", () => {
-  const defaultResizeObserver = global.ResizeObserver
-
```
Comment (108yen) [inline packages/hooks/use-resize-observer/tests/index.test.tsx:7]: Dont remove.

File: packages/hooks/use-resize-observer/tests/index.test.tsx [hunk 4]
```diff
@@ -28,10 +26,6 @@ describe("useResizeObserver", () => {
     }
   })
 
-  afterAll(() => {
-    global.ResizeObserver = defaultResizeObserver
-  })
-
```
Comment (108yen) [inline packages/hooks/use-resize-observer/tests/index.test.tsx:34]: Dont remove.

File: packages/hooks/use-resize-observer/tests/index.test.tsx [hunk 5]
```diff
@@ -45,6 +39,47 @@ describe("useResizeObserver", () => {
...
+            this,
+          )
+        })()
+      }
+      observe = vi.fn()
+      unobserve = vi.fn()
+      disconnect = vi.fn()
+    }
+  })
+
```
Comment (108yen) [inline packages/hooks/use-resize-observer/tests/index.test.tsx:69]: Similarly, use `afterAll` for post-processing.

**Review by 108yen — CHANGES_REQUESTED** [review]

File: .changeset/empty-rules-laugh.md
Comment (kurakke) [reply .changeset/empty-rules-laugh.md:1]: I fixed.
[1d0764b](https://github.com/yamada-ui/yamada-ui/pull/2139/commits/1d0764b2ce9e0add921759cfd1ac4555d671a7cc)

**Comment by kurakke** [comment]
@108yen 
I have made the corrections as per your comments. Please check.

**Review by 108yen — APPROVED** [review]

**Comment by hoshico** [comment]
@kurakke 
thanks for your PR!

The test at line 95 tends to fail after multiple attempts.
Please try running pnpm test use-resize-observer continuously for several dozen times.

If it tends to fail, I recommend using waitFor as referenced in line 48.

**Comment by 108yen** [comment]
@kurakke 
Also, could you please move lines 48 and 95 to `toHaveTextContents`?

**Review by hirotomoyamada — APPROVED** [review]

**Pull request merged**